pkg/spiffe/spiffe.go

```diff
@@ -64,12 +73,49 @@ func SetTrustDomain(value string) {
 	trustDomainMutex.Unlock()
 }
 
+// TODO(incfly): rename this and SetTrustDomain as GetLocalTrustDomain for clarity.
 func GetTrustDomain() string {
 	trustDomainMutex.RLock()
 	defer trustDomainMutex.RUnlock()
 	return trustDomain
 }
 
+func GetTrustDomainByCluster(clusterID string) string {
```

Comment:
Remove thus, see above. Spiffee may have utils to concatenate the strings - but the trust domain is a MeshConfig setting, and it can't work if each cluster defines different trust domains - pods from different clusters must communicate with each other.

Reply:
done.